### ISSUE TITLE
The network on pytorch and on leelaz have different outputs.

### ISSUE THREAD

**Issue by CGLemon** [yukw777/leela-zero-pytorch#2]
I want to train a  9x9 network. After trianing the network, I compare the weight on  pytorch and on leelaz. I can not get the same value and policy, but leelaz seem to move correctly. I don't  know what happened. Need some help. Thanks!

**Comment by yukw777** [comment]
I don't quite understand what you mean. Do you mean that when you feed in the same board position with the same history to your 9x9 model in PyTorch and Leelaz, you don't get the same value and policy?

**Comment by yukw777** [comment]
Also, have you compiled leela zero with the appropriate board size?

https://github.com/leela-zero/leela-zero/blob/b259e50d5cce34a12176846534f369ef5ffcebc1/src/config.h#L48

**Comment by CGLemon** [comment]
Yes, I change  board size to 9 in Leelaz and your tool. It can generate right model and Leelaz accept the model  .But when I feed in the same board position with the same history to PyTorch and Leelaz, I don't get the same value and policy.

**Comment by yukw777** [comment]
hmm. let's dive into a bit deeper.. Can you tell me the detailed steps to reproduce? I haven't actually verified that the pytorch model and leela zero actually produce the same value and policy as i just assumed as long as my weight format is correct, it'd work. Thanks for doing this check! Maybe there's a bug lurking around..

**Comment by CGLemon** [comment]
I change board size to 9  and generate a sample weight, named 'network.ckpt' and 'network.txt'.

Before testing the network, we need to recode Leelaz in order to get the net outputs. In the source file 'Network.cpp', the function 'Network::Netresult Network::get_output_internal' generate input pattens, policy and value. The 'input_data' is the input pattens. The 'winrate_out' is the value head output without tanh. The 'policy_out' is the policy head output without softmax.

Leelaz read the  'network.txt' with command ' ./leelaz -w network.txt -p 1 --noponder -t 1 --batchsize 1 '. We restrict threads in one, so we can get input and outputs one by one. Copy the input vector to PyTorch model. Now we can compare the PyTorch outputs  and Leelaz outputs.

**Comment by CGLemon** [comment]
I find a problem with 
https://github.com/yukw777/leela-zero-pytorch/blob/72024b36aaf0fb0853cb7d6ce3ab8cd17f6adc12/leela_zero_pytorch/network.py#L184

'bias = conv_block.beta * torch.sqrt(conv_block.bn.running_var + conv_block.bn.eps)' is correct.

But I still can not get the same output.

**Comment by CGLemon** [comment]
I found that the output from PyTorch and from Leelaz are always diffrerent. Because running_var and running_mean only exist in training time, they are not  written in .ckpt  file. Absolutely we will not get the same value and policy.

**Comment by CGLemon** [comment]
After using PyTorch model function 'model.eval()'. I can get the same value and policy. Thank you vary much.

**Comment by yukw777** [comment]
@CGLemon Thanks for looking into it further! Glad it was something simple.
